Fix sheet reset

mkbel opened the pull request.

mkbel committed:
Remove statements with no effect.

mkbel committed:
Change default sheet names.

mkbel committed:
Extract function next_initialized_sheet from change_sheet_from_menu(). Make it more general to depend on sizeof(Sheets::s) instead of hard coded value of 2.

mkbel committed:
Move sheet menu up to be after Print from SD.

mkbel committed:
Hide sheet switcher from main menu, if there are less than 2 sheets calibrated.

mkbel committed:
Reset calibration status if there is no calibrated sheet after reset.

mkbel committed:
Increase number of configurable steel sheets to 8. Move sheets into sub menu of HW setup.

mkbel committed:
Migrate old vale of EEPROM_BABYSTEP_Z only if active sheet is not initialized.

mkbel committed:
Document

mkbel committed:
Convert sheet switcher from sub menu to function. Handle null characters in sheet name.

mkbel committed:
Fix sheet menu back caption to Steel sheets to match parent menu caption.

mkbel committed:
Convert reset sheet sub menu to function.

vintagepc commented:
Argh! you beat me on the menu move in #2146 :stuck_out_tongue:

mkbel committed:
Use EEPROM update instead of write.

DRracer commented:
@vintagepc yeah, more people were asking to see the selected sheet somewhere above in the menu

mkbel committed:
Fix compiler warning.

mkbel committed:
Switch to next sheet if currently selected sheet is not initialized. Move next_initialized_sheet() and change_sheet_from_menu() to eeprom.cpp, rename it to eeprom_next_initialized_sheet() and eeprom_switch_to_next_sheet(). Rename default_sheet_name() to eeprom_default_sheet_name().

mkbel committed:
Do not activate selected_sheet in lcd_v2_calibration() as this leads to calibrating random sheet if activated from calibration menu.

selected_sheet is initialized to 0 and contains index of last Sheet menu visited.

mkbel committed:
Change default sheet name from "<number>" to "Sheet <number>".
